django_log_formatter_asim/__init__.py

```diff
@@ -1,176 +1,126 @@
-            "env": self._get_environment(),
+        if not getattr(settings, "DLFA_LOG_PERSONALLY_IDENTIFIABLE_INFORMATION", False):
+            self._replace_personally_identifiable_information()
+
+    def _replace_personally_identifiable_information(self):
+        if getattr(self.record, "request", None):
+            user = getattr(self.record.request, "user", None)
+            if user:
+                user.username = "{{USERNAME}}"
+                user.email = "{{EMAIL}}"
+                user.first_name = "{{FIRST_NAME}}"
+                user.last_name = "{{LAST_NAME}}"
+
+    def get_log_dict_base(self):
+        record = self.record
+        log_time = datetime.utcfromtimestamp(record.created).isoformat()
+        log_dict = {
+            # Event fields...
+            "EventMessage": record.msg,
+            "EventCount": 1,
+            "EventStartTime": log_time,
+            "EventEndTime": log_time,
+            "EventType": record.name,
+            "EventResult": "NA",
+            "EventSeverity": self._get_event_severity(record.levelname),
+            "EventOriginalSeverity": record.levelname,
+            "EventSchema": "ProcessEvent",
+            "EventSchemaVersion": "0.1.4",
+            "ActingAppType": "Django",
+            # Other fields...
+            "AdditionalFields": {
+                "DjangoLogFormatterAsimVersion": distribution("django-log-formatter-asim").version,
+                "TraceHeaders": {},
+            },
         }
 
-        logger = (
-            ASIMLogger()
-            .event(
-                category=self._get_event_category(),
-                action=self.record.name,
-                message=self.record.getMessage(),
-                labels={
-                    **labels,
-                    **extra_labels,
-                },
-            )
-            .host(
-                architecture=platform.machine(),
-            )
-        )
+        if getattr(settings, "DLFA_INCLUDE_RAW_LOG", False):
+            log_dict["AdditionalFields"]["RawLog"] = json.dumps(record, default=self._to_dict)
 
-        return logger
+        return log_dict
 
-    def _get_event_category(self):
-        if self.record.name in ("django.request", "django.server"):
-            return CATEGORY_WEB
-        if self.record.name.startswith("django.db.backends"):
-            return CATEGORY_DATABASE
+    def _to_dict(self, object):
+        try:
+            return vars(object)
+        except TypeError:
+            return str(object)
 
-        return CATEGORY_PROCESS
-
-    def _get_environment(self):
-        return os.getenv("DJANGO_SETTINGS_MODULE") or "Unknown"
+    def _get_event_severity(self, log_level):
+        map = {
+            "DEBUG": "Informational",
+            "INFO": "Informational",
+            "WARNING": "Low",
+            "ERROR": "Medium",
+            "CRITICAL": "High",
+        }
+        return map[log_level]
 
 
 class ASIMSystemFormatter(ASIMFormatterBase):
```

Comment:
Is there any benefit in having `ASIMSystemFormatter`?

I feel like we could just rename `ASIMFormatterBase` to `ASIMFormatter` and use that as the default, extending it as required, e.g. `ASIMRequestFormatter`.